src/Match2Cases.jl

```diff
@@ -87,10 +87,15 @@ function generate_code(top_down_states::Vector{DeduplicatedCodePoint}, value, lo
     reverse!(top_down_states)
     togen = top_down_states # but now it's in the right order to consume from the end
 
+    labels = IdDict{DeduplicatedCodePoint, Symbol}()
+    function label(code::DeduplicatedCodePoint)
+        get!(() -> gensym("label", state), labels, code)
+    end
+
     while !isempty(togen)
         pc = pop!(togen)
-        if pc.label isa Symbol
-            push!(emit, :(@label $(pc.label)))
+        if pc in keys(labels)
```

Comment:
nifty!

this works because we are guaranteed to emit the `@goto` before we hit the node where we need the `@label`, right?

Reply:
if so, could you add a comment to that effect maybe before line 90?

Reply:
Yes.  That is because the nodes are topologically sorted.